File: src/main/scala/acceptanceServer/ServerJudge.scala
package acceptanceServer

import java.io.IOException
import java.net.Socket

object ServerJudge {
  private val address: String = ConnectionSetting.SERVER_JUDGE_ADDRESS
  private val port: Int = ConnectionSetting.SERVER_JUDGE_PORT
}

class ServerJudge() {
  while (true) {
    {
      try {
        conn = new acceptanceServer.Client(new Socket(ServerJudge.address, ServerJudge.port))
        server_score = new ServerScore
      }
      catch {
        case e: IOException => {
          System.out.print(e)
          continue //todo: continue is not supported
        }
      }
      try {
        conn.openStream()
        break //todo: break is not supported
      }
      catch {
        case e: IOException => {
          e.printStackTrace()
          continue //todo: continue is not supported
        }
      }
    }
  }
  private var conn: acceptanceServer.Client = null
  private var server_score: ServerScore = null

  def judge(host: acceptanceServer.Player, players: util.ArrayList[acceptanceServer.Player]): util.ArrayList[Integer] = {
    val host_score: Int = server_score.score(host.getCard)
    val list: util.ArrayList[Integer] = new util.ArrayList[Integer]
    var i: Int = 0
    while (i < players.size) {
      {
        val hand: util.ArrayList[Card] = players.get(i).getCard
        val player_score: Int = server_score.score(hand)
        conn.println(host_score + " " + player_score)
        var reply: String = ""
        while (reply == "") {
          {
            reply = conn.read
          }
        }
        list.add(reply.toInt)
      }
      {
        i += 1; i - 1
      }
    }
    return list
  }
}

File: src/serverApp/ServerJudge.java
package serverApp;

import gameServer.Player;
import gameServer.ServerScore;

import java.io.IOException;
import java.net.Socket;
import java.util.ArrayList;

public class ServerJudge {
	private static final String address = ConnectionSetting.SERVER_JUDGE_ADDRESS;
	private static final int port = ConnectionSetting.SERVER_JUDGE_PORT;
	private acceptanceServer.Client conn = null;
	private gameServer.ServerScore server_score = null;
	
	public ServerJudge(){		
		while(true){
			try{	
				conn = new acceptanceServer.Client(new Socket(address, port));
				server_score = new ServerScore();
			} catch(IOException e) {
				System.out.print(e);
				continue;
			}
		
			try{
				conn.openStream();
				break;
			} catch(IOException e) {
				e.printStackTrace();
				continue;
			}
		}
	}
	
	public ArrayList<Integer> judge(gameServer.Player host, ArrayList<Player> players){
		int host_score = server_score.score(host.getCard());
		ArrayList<Integer> list = new ArrayList<Integer>();
		
		for(int i = 0; i < players.size(); i++){
			ArrayList<gameServer.Card> hand = players.get(i).getCard();
			int player_score = server_score.score(hand);
			conn.println(host_score + " " + player_score);
			
			String reply = "";
			while(reply.equals("")){
				reply = conn.read();
			}
			list.add(Integer.parseInt(reply));
		}
				
		return list;
	}
}
